adagrams/game.py

```diff
@@ -1,11 +1,155 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+
+    final_string = []
+    letters_frequency = []
+    letters_list = []
+    
+    for letters in LETTER_POOL:
+        letters_frequency.append(LETTER_POOL[letters])
+    
+
+    for letter in LETTER_POOL:
+        letters_list.append(letter)
+
+    while len(final_string) < 10:
+        letter_index = random.randint(0, len(LETTER_POOL) -1)
+
+        if letters_frequency[letter_index] > 0:
+            letters_frequency[letter_index] -= 1
+            final_string.append(letters_list[letter_index])
+        
+    return final_string
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    available_letters = []
+
+
+    for letter_item in letter_bank:
+        available_letters.append(letter_item)
+
+    for letter in word:
+        uppercase_letter = letter.upper()
+        if uppercase_letter in available_letters:
+            
+            available_letters.remove(uppercase_letter)
+        else: 
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    
+    score_values = {
+        1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
```

Comment:
Since we know that these values aren't changing we could use a tuple here instead of list. Tuples are data types that can't be change and so we also ensure that they aren't accidentally changed when working with our function. Also since `score_values` isn't being modified should we consider naming it `SCORE_VALUES` and globally initializing it?